service/file_transaction_service.py

```diff
@@ -86,20 +87,29 @@ def CreateUserFolder(self, request, context):
         is_uuid_valid = utility.verify_uuid(request.uuid)
 
         if is_uuid_valid:
-            count = utility.count_folders(request.uuid)
-            logger.info("count:", count)
-            created = utility.create_uuid_container_in_azure(count, request.uuid)
+            # Lock this uuid
+            self.__server.get_uuid_locker()[request.uuid].acquire()
 
-            if created:
-                return hwsc_file_transaction_svc_pb2.FileTransactionResponse(
-                    code=grpc.StatusCode.OK.value[0],
-                    message="success"
-                )
-            else:
-                return hwsc_file_transaction_svc_pb2.FileTransactionResponse(
-                    code=grpc.StatusCode.UNKNOWN.value[0],
-                    message="user folder already exist"
-                )
+            try:
+                if not azure_client.user_folders_exist_in_azure(request.uuid):
+                    azure_client.create_uuid_container_in_azure(request.uuid)
+                    context.set_code = grpc.StatusCode.OK.value[0]
+                    context.set_details = "user folder creation successful"
+                else:
+                    context.set_code = grpc.StatusCode.UNKNOWN.value[0]
+                    context.set_details = "user folder already exists"
+            except Exception as e:
+                logger.exception(str(e))
```

Comment:
I am curious why the casting to `str` `logger.exception(str(e))`

Reply:
`e` is cast to `str` because of how I implemented `exception` in `logger.py` to be consistent with the other logger functions. Should we just use the default `exception` in the `logging` module?